distributed/comm/addressing.py

```diff
@@ -119,15 +119,24 @@ def get_address_host(addr):
 
     In contrast to get_address_host_port(), this function should always
     succeed for well-formed addresses.
-
-    >>> get_address_host('tcp://192.168.10.50:8687')
-    '192.168.10.50'
     """
     scheme, loc = parse_address(addr)
     backend = registry.get_backend(scheme)
     return backend.get_address_host(loc)
 
 
+def get_local_address_for(addr):
+    """
+    Get the local listening address suitable for reaching *addr*.
+
+    For instance, trying to reach an external TCP address will return
+    a local TCP address that's routable to that external address.
+    """
```

Comment:
I'm not very happy with the wording either... It's a generic equivalent of `get_ip()`, roughly:
```
>>> get_local_address_for('tcp://8.8.8.8:1234')
'tcp://192.168.1.68'
>>> get_local_address_for('tcp://127.0.0.1:1234')
'tcp://127.0.0.1'
```

This won't ever try to choose another address or change transports. When we want to do that, we'll have to define another API.